packages/react-reconciler/src/ReactFiberClassComponent.new.js

```diff
@@ -184,12 +185,6 @@ function applyDerivedStateFromProps(
       }
     }
   }
-
-  const partialState = getDerivedStateFromProps(nextProps, prevState);
-
-  if (__DEV__) {
-    warnOnUndefinedDerivedState(ctor, partialState);
-  }
```

Comment:
Ah, so this PR does more than just disable logs during the second pass. It also changes the code to _use_ the result of the first pass rather than the second pass? Should be okay so long as there are no unsafe side effects?

I think that using the 2nd call (for this and other methods below) may have been an intentional choice _in case_ there were side effects.

Reply:
Hmm. I haven't thought about this. I do vaguely remember some two year old conversations about this. Maybe let's hold off merging and keep it for the next sync discussion.

Reply:
Cool beans 👍🏼

Reply:
So the double rendering order in general is very intentional eg renderWithHooks.

However I noticed when doing the Fizz parts that some of the class ones are reverse order. I'm not sure that was intentional or an oversight when we intentionally did the other ones.

Because I think there are some subtle semantic issues with things like memoization in general that's an issue here. Deduping warnings being one such memoization that breaks.

Reply:
The way we do it with Hooks is we disable logs during the second one _and_ use the result of the second one. I could make everything match tha.

Reply:
I've been extremely confused before why the logs and the returned values didn't match... Maybe disabling the logs on the first call and enable logs and use the result of the second call?